### DISCUSSION TITLE
A way to replace/override the onRowClick handler

### DISCUSSION THREAD

**Discussion by maxacarvalho** [directus/directus#7518] [Archive]
I'm trying to find a way to customize/overwrite the `onRowClick` method, any thoughts?
Thanks!

**Comment by rijkvanzanten** [comment]
What onRowClick method where?

**Reply by maxacarvalho** [reply]
Hi @rijkvanzanten 

Sorry for the lack of reference.  
I'm talking about the `onRowClick` on the file `app/src/layouts/tabular/index.ts`. That method is sent to the `tabular.vue` file that then, sends it to the `v-table` component.

I'd like to have a custom module that changes the behavior of that method, so it performs a different action instead.  

My current use-case is: I have a custom module that adds new routes. The default `onRowClick` method navigates to `/collections/${collection.value}/${encodeURIComponent(primaryKey)}`. My module has its own routes, so to list a collection it would be something like `/module/custom_name/`, then, to open an item it would be `/module/custom_name/12345` (where `12345` is the ID of the item).

That's one use case, but in general, I believe it would be nice to have a way to easily override that method, so custom modules can add their own functionality out of the row click action.

**Reply by rijkvanzanten** [reply]
Are you using the existing tabular layout in your module? 

The tabular layout is meant to either select or navigate to the detail pages. I'd recommend relying on the regular `v-table` component if you have the need for any custom logic for the rows 👍🏻

**Reply by maxacarvalho** [reply]
Hummmm, yes, I'm re-using the tabular layout, for practical reasons, but you are right, I can skip that and use the `v-table` directly. Actually, while searching for the `onRowClick` reference I found the one from the file `app/src/modules/settings/routes/presets/collection/collection.vue`, which seems to be exactly what I want. I believe I can use that as a reference.

**Reply by rijkvanzanten** [reply]
Yeah exactly! Note how that's also using the regular `v-table` component, instead of the layout 🙂 

The layouts are designed to work under the collections module, so as soon as you take them out of that context and start tweaking functionality you might run into unexpected behavior.

https://github.com/directus/directus/blob/e1cbfc4ca4a9f4a0de6b9c0220b87283b43d5d06/app/src/modules/settings/routes/presets/collection/collection.vue#L52-L78

**Reply by maxacarvalho** [reply]
Got it, that's perfect @rijkvanzanten, thanks for your help!!!!
